src/overlay-tools/SHIPBroadcaster.ts

```diff
@@ -0,0 +1,405 @@
+  /** Determines a mapping whose keys are specific hosts and whose values are the topics (all, any, or a specific list) that must be present within the STEAK received by the given hosts, in order for the broadcast to be considered a success. */
+  requireAcknowledgmentFromSpecificHostsForTopics?: Record<string, 'all' | 'any' | string[]>
+}
+
+/** Facilitates transaction broadcasts that return STEAK. */
+export interface OverlayBroadcastFacilitator {
+  send: (url: string, taggedBEEF: TaggedBEEF) => Promise<STEAK>
+}
+
+export class HTTPSOverlayBroadcastFacilitator implements OverlayBroadcastFacilitator {
+  httpClient: typeof fetch
+
+  constructor(httpClient = fetch) {
+    this.httpClient = httpClient
+  }
+
+  async send(url: string, taggedBEEF: TaggedBEEF): Promise<STEAK> {
+    if (!url.startsWith('https:')) {
+      throw new Error('HTTPS facilitator can only use URLs that start with "https:"')
+    }
+    const response = await fetch(`${url}/submit`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/octet-stream',
+        'X-Topics': JSON.stringify(taggedBEEF.topics)
+      },
+      body: new Uint8Array(taggedBEEF.beef)
+    })
+    if (response.ok) {
+      return await response.json()
+    } else {
+      throw new Error('Failed to facilitate broadcast')
+    }
+  }
+}
+
+/**
+ * Represents a SHIP transaction broadcaster.
+ */
+export default class SHIPCast implements Broadcaster {
+  private readonly topics: string[]
+  private readonly facilitator: OverlayBroadcastFacilitator
+  private readonly resolver: LookupResolver
+  private readonly requireAcknowledgmentFromAllHostsForTopics: 'all' | 'any' | string[]
+  private readonly requireAcknowledgmentFromAnyHostForTopics: 'all' | 'any' | string[]
+  private readonly requireAcknowledgmentFromSpecificHostsForTopics: Record<string, 'all' | 'any' | string[]>
+
+  /**
+     * Constructs an instance of the SHIP broadcaster.
+     *
+     * @param {string[]} topics - The list of SHIP topic names where transactions are to be sent.
+     * @param {SHIPBroadcasterConfig} config - Configuration options for the SHIP broadcaster.
+     */
+  constructor(topics: string[], config?: SHIPBroadcasterConfig) {
+    if (topics.length === 0) {
+      throw new Error('At least one topic is required for broadcast.')
+    }
+    if (topics.some(x => !x.startsWith('tm_'))) {
+      throw new Error('Every topic must start with "tm_".')
+    }
+    this.topics = topics
+    const {
+      facilitator, resolver,
+      requireAcknowledgmentFromAllHostsForTopics,
+      requireAcknowledgmentFromAnyHostForTopics,
+      requireAcknowledgmentFromSpecificHostsForTopics
+    } = config ?? {} as SHIPBroadcasterConfig
+    this.facilitator = facilitator ?? new HTTPSOverlayBroadcastFacilitator()
+    this.resolver = resolver ?? new LookupResolver()
+    this.requireAcknowledgmentFromAllHostsForTopics = requireAcknowledgmentFromAllHostsForTopics ?? 'all'
```

Comment:
The default should be set so that ACK from any host for each topic is acceptable. So 

```suggestion
    this.requireAcknowledgmentFromAllHostsForTopics = requireAcknowledgmentFromAllHostsForTopics ?? []
```